src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -137,14 +156,60 @@ public boolean send(final EventBatch batch) {
     @Override
     public synchronized String executeHttpRequest(final HttpUriRequest req) {
         CloseableHttpResponse resp;
-        try {
-            resp = httpClient.execute(req, context);
-        } catch (Exception ex) {
-            logBackPressure();
-            log.error("encountered io exception", ex);
-            throw new HecException("encountered exception when post data", ex);
+        if (hecConfig.kerberosAuthEnabled()) {
+            if (config == null) {
+                config = new Configuration() {
+                    @SuppressWarnings("serial")
+                    @Override
+                    public AppConfigurationEntry[] getAppConfigurationEntry(String name) {
+                        return new AppConfigurationEntry[]{new AppConfigurationEntry("com.sun.security.auth.module.Krb5LoginModule",
+                            AppConfigurationEntry.LoginModuleControlFlag.REQUIRED, new HashMap<String, Object>() {
+                            {
+                                put("useTicketCache", "false");
+                                put("useKeyTab", "true");
+                                put("keyTab", hecConfig.kerberosKeytabLocation());
+                                //Krb5 in GSS API needs to be refreshed so it does not throw the error
+                                //Specified version of key is not available
+                                put("refreshKrb5Config", "true");
+                                put("principal", hecConfig.kerberosPrincipal());
+                                put("storeKey", "false");
+                                put("doNotPrompt", "true");
+                                put("isInitiator", "true");
+                                put("debug", "true");
+                            }
+                        })};
+                    }
+                };
+            }
+            Set<Principal> princ = new HashSet<Principal>(1);
+            princ.add(new KerberosPrincipal(hecConfig.kerberosUser()));
+            Subject sub = new Subject(false, princ, new HashSet<Object>(), new HashSet<Object>());
+            try {
+                LoginContext lc = new LoginContext("", sub, null, config);
+                lc.login();
+                Subject serviceSubject = lc.getSubject();
+                resp = Subject.doAs(serviceSubject, new PrivilegedAction<CloseableHttpResponse>() {
+                    @Override
+                    public CloseableHttpResponse run() {
+                        try {
+                            return httpClient.execute(req, context);
+                        } catch (IOException ex) {
+                            logBackPressure();
```

Comment:
Need to know what this does...

Reply:
the logBackPressure() increases the backPressure if the execute fails. Based on the threshold for backPressure, it delays the call of send function for other batches into the channel